src/dotenv.ts

```diff
@@ -53,7 +71,7 @@ export async function setupDotenv(options: DotenvOptions): Promise<Env> {
 
   // Fill process.env
   for (const key in environment) {
-    if (!key.startsWith("_") && targetEnvironment[key] === undefined) {
+    if (!key.startsWith("_")) {
```

Comment:
Can we add `=== undefined` condition? (we can add `|| registry.has(key)` allow overide existing ones)